community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/commands/expressions/PointFunction.scala

```diff
@@ -27,14 +27,28 @@ import org.neo4j.cypher.internal.runtime.interpreted.IsMap
 import org.neo4j.cypher.internal.runtime.interpreted.pipes.QueryState
 import org.neo4j.values.AnyValue
 import org.neo4j.values.storable.{PointValue, Values}
-import org.neo4j.values.virtual.MapValue
+import org.neo4j.values.virtual.{MapValue, VirtualNodeValue, VirtualRelationshipValue, VirtualValues}
+
+import scala.collection.JavaConverters._
 
 case class PointFunction(data: Expression) extends NullInNullOutExpression(data) {
   override def compute(value: AnyValue, ctx: ExecutionContext, state: QueryState): AnyValue = value match {
     case IsMap(mapCreator) =>
       val map = mapCreator(state.query)
-      if (containsNull(map)) Values.NO_VALUE
-      else PointValue.fromMap(map)
+      if (containsNull(map)) {
+        Values.NO_VALUE
+      } else {
+        if (value.isInstanceOf[VirtualNodeValue] || value.isInstanceOf[VirtualRelationshipValue]) {
```

Comment:
This code could have been implemented inside of PointValue.fromMap, although that would have required the map filter to be done in java.

Reply:
Also, fromMap is called from the CSV parser as well. And there we don't want this behavior.